### PR TITLE
fix(verticalNav): update state of selected nav item

### PR DESCRIPTION
**Pull request by varnikajain15**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

**Review by Rashi-Gupta-2000 — COMMENTED** [review]
Please revisit the changes in the MenuItem file as the states are breaking in some places like the pill in disabled state - 
Screenshot_2023-01-18_12-57-03

**Review by Rashi-Gupta-2000 — COMMENTED** [review]
Please change the commit message to `update states of selected nav item`

**Review by Rashi-Gupta-2000 — COMMENTED** [review]
The color of the arrow should be according to the color of the text as confirmed by the designer 
Please change the color of the arrow in case of selected active states 
Screenshot_2023-01-18_13-28-03

**Review by Rashi-Gupta-2000 — COMMENTED** [review]
Please remove the focus state of disabled 
Screenshot_2023-01-18_13-31-07

**Review by Rashi-Gupta-2000 — COMMENTED** [review]
Please check for focus ring on selected item and active state

**Review by Rashi-Gupta-2000 — COMMENTED** [review]
Please check that the title and commit message need to be the same.

File: core/components/organisms/verticalNav/MenuItem.tsx [hunk 1]
```diff
@@ -23,34 +24,25 @@ interface MenuPillsProps {
...
 
 interface MenuIconProps {
   isChildrenVisible?: boolean;
-  isActive?: boolean;
-  disabled?: boolean;
 }
 
 const MenuLabel = (props: MenuLabelProps) => {
-  const { label, disabled, isActive } = props;
+  const { label, labelColor } = props;
```
Comment (Rashi-Gupta-2000) [inline core/components/organisms/verticalNav/MenuItem.tsx:35]: Please add test cases for the new prop added.

File: core/components/organisms/verticalNav/MenuItem.tsx [hunk 2]
```diff
@@ -98,11 +92,12 @@ export const MenuItem = (props: MenuItemProps) => {
     ['MenuItem--disabled']: menu.disabled,
     ['MenuItem--subMenu']: isChildren && expanded,
     ['MenuItem--rounded']: rounded && expanded,
+    [`color-${itemColor}`]: true,
```
Comment (Rashi-Gupta-2000) [inline core/components/organisms/verticalNav/MenuItem.tsx:95]: Please check if we can add test cases for this class.

Comment (varnikajain15) [reply core/components/organisms/verticalNav/MenuItem.tsx:95]: I have added test case for this class already.

File: core/components/organisms/verticalNav/MenuItem.tsx (hunk 1 above)
Comment (varnikajain15) [reply core/components/organisms/verticalNav/MenuItem.tsx:35]: This prop passes the color to the label. So, I have added test case to check the color of the label in different states.

**Review by Rashi-Gupta-2000 — APPROVED** [review]

**Review by anuradha9712 — APPROVED** [review]

**Pull request merged**